pkg/app/server/grpcapi/api.go

```diff
@@ -343,6 +345,30 @@ func (a *API) ListApplications(ctx context.Context, req *apiservice.ListApplicat
 	}, nil
 }
 
+func (a *API) UpdateApplication(ctx context.Context, req *apiservice.UpdateApplicationRequest) (*apiservice.UpdateApplicationResponse, error) {
+	key, err := requireAPIKey(ctx, model.APIKey_READ_WRITE, a.logger)
+	if err != nil {
+		return nil, err
+	}
+
+	app, err := getApplication(ctx, a.applicationStore, req.ApplicationId, a.logger)
+	if err != nil {
+		return nil, err
+	}
+
+	if app.ProjectId != key.ProjectId {
+		return nil, status.Error(codes.InvalidArgument, "Requested application does not belong to your project")
+	}
+
+	if err := a.applicationStore.UpdateConfiguration(ctx, req.ApplicationId, app.PipedId, req.PlatformProvider, req.GitPath.ConfigFilename); err != nil {
```

Comment:
Instead of `app.PipedId` should be `req.PipedId`. I mean, you can even change the piped which handles this application via this grpc, so why not?
Just as the same we do in web_api.go (ref: [code](https://github.com/pipe-cd/pipecd/blob/master/pkg/app/server/grpcapi/web_api.go#L530-L551))